opengever/maintenance/scripts/sql_contact_migration.py

```diff
@@ -48,6 +57,49 @@ def export(self):
         org_roles = list(self.get_org_roles())
         self.export_json('memberships.json', org_roles)
 
+    def migrate_participations(self):
+        catalog = api.portal.get_tool('portal_catalog')
+        brains = catalog.unrestrictedSearchResults(
+            object_provides=IDossierMarker.__identifier__)
+        for brain in brains:
+            dossier = brain.getObject()
+            self.migrate_participation(dossier)
+
+    def migrate_participation(self, dossier):
+        sql_handler = SQLParticipationHandler(dossier)
+        kub_handler = KuBParticipationHandler(dossier)
+        participations = sql_handler.get_participations()
+
+        for participation in participations:
+            # To avoid reindexing the dosiser after each participation we
+            # add the participation manually
+            if isinstance(participation, OgdsUserParticipation):
```

Comment:
The `IParticipationData` unfortunately only returns the string with the GEVER contact_id but I need the newly generated kub_id. And because the storage of persons and organization are different than for OrgRoles its easier to have some ifs 😓

Reply:
ah yes, sorry, I did not read carefully the other conditions 😓